contracts/FastBreak.cdc

```diff
@@ -0,0 +1,682 @@
+                self.fatigueModeOn = fastBreakRun.fatigueModeOn
+            } else {
+                self.id = id
+                self.name = name
+                self.status = "FAST_BREAK_RUN_OPEN"
+                self.runStart = runStart
+                self.runEnd = runEnd
+                self.leaderboard = {}
+                self.fatigueModeOn = fatigueModeOn
+            }
+        }
+
+        /// Update status of the Fast Break Run
+        ///
+        access(contract) fun updateStatus(status: String) { self.status = status }
+
+        /// Write a new win to the Fast Break Run leaderboard
+        ///
+        access(contract) fun incrementLeaderboardWins(wallet: Address) {
+            let leaderboard = self.leaderboard
+            var wins = leaderboard[wallet] ?? 0
+            wins = wins + 1
+            leaderboard[wallet] = wins
+        }
+    }
+
+    /// Get a Fast Break Run by Id
+    ///
+    pub fun getFastBreakRun(id: String): FastBreak.FastBreakRun? {
+        return FastBreak.fastBreakRunByID[id]
+    }
+
+    /// A single Game of Fast Break
+    ///
+    pub struct FastBreakGame {
+        pub let id: String
+        pub let name: String
+        pub let isPublic: Bool
+        pub let submissionDeadline: UInt64
+        pub let numPlayers: UInt64
+        pub var status: String
+        pub var winner: Address?
+        pub var submissions: {Address: FastBreak.FastBreakSubmission}
+        pub let fastBreakRunID: String
+        pub var stats: [FastBreakStat]
+
+        init (
+            id: String,
+            name: String,
+            fastBreakRunID: String,
+            isPublic: Bool,
+            submissionDeadline: UInt64,
+            numPlayers: UInt64
+        ) {
+            if let fb = FastBreak.fastBreakGameByID[id] {
+                self.id = fb.id
+                self.name = fb.name
+                self.isPublic = fb.isPublic
+                self.submissionDeadline = fb.submissionDeadline
+                self.numPlayers = fb.numPlayers
+                self.status = fb.status
+                self.winner = fb.winner
+                self.submissions = fb.submissions
+                self.fastBreakRunID = fb.fastBreakRunID
+                self.stats = fb.stats
+            } else {
+                self.id = id
+                self.name = name
+                self.isPublic = isPublic
+                self.submissionDeadline = submissionDeadline
+                self.numPlayers = numPlayers
+                self.status = "FAST_BREAK_OPEN"
+                self.submissions = {}
+                self.fastBreakRunID = fastBreakRunID
+                self.stats = []
+                self.winner = nil
+            }
+        }
+
+        /// Get a wallet's active Fast Break Submission
+        ///
+        pub fun getFastBreakSubmissionByWallet(wallet: Address): FastBreak.FastBreakSubmission? {
+            let fastBreakSubmissions = self.submissions!
```

Comment:
can you explain why force-unwrap is bad?

Reply:
If the value that you're trying to unwrap is `nil`, then the force-unwrap will fail and the entire transaction execution will also fail with an unhelpful error message about the force unwrap failing, but no context. Force unwrap should really only be used when you are 100% sure that the optional will never be `nil`. 

Alternatively, you should use if-let to gracefully unwrap the optional and if it is `nil`, you can handle that without having to fail the entire transaction. This example, a getter function, should just return `nil` if `self.submissions` is `nil` to indicate that the submissions was `nil`. As a last resort, if you have to revert the transaction if the value is `nil`, you should just `panic()` with a descriptive error message, because that is much more helpful than just failing with none

Reply:
@HouseOfHufflepuff reminder about this